Xamarin.Forms.Core/Device.cs

```diff
@@ -104,6 +104,85 @@ public static void BeginInvokeOnMainThread(Action action)
 			PlatformServices.BeginInvokeOnMainThread(action);
 		}
 
+		public static Task InvokeOnMainThreadAsync(Action action)
+		{
+			var tcs = new TaskCompletionSource<object>();
+			BeginInvokeOnMainThread(() =>
+			{
+				try
+				{
+					action();
+					tcs.SetResult(null);
+				}
+				catch (Exception e)
+				{
+					tcs.SetException(e);
+				}
+			});
+
+			return tcs.Task;
+		}
+
+		public static Task InvokeOnMainThreadAsync(Task actionTask)
```

Comment:
The task is only awaited on the main thread but not started on it. Is this the intended behaviour?